Supplyment multicast document with a limitation (#4850)
antrea-io/antrea: docs/multicast-guide.md

Configuration option `multicastInterfaces` is not supported with encap mode.
Multicast packets in encap mode are SNATed and forwarded to the transport interface only.

> ### Large number of receiver groups for one node

Reviewer: ```suggestion
### Large number of receiver groups for one Node
```

Author: updated

> ### Large number of receiver groups for one node

Reviewer 2: "Maximum number of receiver groups on one Node"?

Author: updated

A Linux host limits the maximum number of multicast groups it can subscribe to

> by the number of [/proc/sys/net/ipv4/igmp_max_memberships](https://sysctl-explorer.net/net/ipv4/igmp_max_memberships/).

Reviewer 3: Append "The default value is 20." after this sentence.

Author: updated

> A Linux host limits the maximum number of multicast groups it can subscribe to
> by the default number of 20 [/proc/sys/net/ipv4/igmp_max_memberships](https://sysctl-explorer.net/net/ipv4/igmp_max_memberships/).

Reviewer: Better to say:
```suggestion
A Linux host limits the maximum number of multicast groups it can subscribe to, 
the default number is 20 [/proc/sys/net/ipv4/igmp_max_memberships](https://sysctl-explorer.net/net/ipv4/igmp_max_memberships/).
```

Author: I think your suggestion is grammatically incorrect. `A Linux host limits the maximum number of multicast groups it can subscribe to, 
the default number is 20`  contains two independent clauses, and they are improperly joined with a comma.

Reviewer 2: Maybe: "A Linux host limits the maximum number of multicast groups it can subscribe to; the default number is 20. The limit can be changed by setting [/proc/sys/net/ipv4/igmp_max_memberships](https://sysctl-explorer.net/net/ipv4/igmp_max_memberships/)."?

Reviewer: @ceclinux you can follow @jianjuns 's suggestion to refine the sentence, the purpose is to make the sentence more readable.

Author: updated. Thanks

> Users should change this number according to their needs, and Antrea is not

Reviewer 3: ```suggestion
Users are responsible to change the value if more than 20 multicast groups are expected per Node.
```

Author: updated. I think the correct wording should be "responsible for" https://www.britannica.com/dictionary/eb/qa/What-preposition-should-be-used-after-responsible-

> Users should change this number according to their needs, and Antrea is not

Author: updated

> Users are responsible for changing the value if Pods on this Node are expected to

Reviewer: ```suggestion
Users are responsible to change the value if Pods on the Node are expected to
```

Reviewer 2: "for changing" is correct.

Reviewer 2: Maybe change "the value" to "the limit".

Author: updated
